File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int out = 0, l = 0, maxf = 0;
        map<char, int> hash;
        for(int r = 0; r < s.size(); r++)
        {
            hash[s[r]]++;
            maxf = max(maxf, hash[s[r]]);

            while((r - l + 1) - maxf > k)
            {
                hash[s[l]]--;
                l++;
            }
            out = max(out, r - l + 1);
        }
        return out;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cs
public class Solution {
    public int CharacterReplacement(string s, int k)
    {
        int l = 0, r = 0, result = 0;
        var dic = new Dictionary<char, int>();

        while (r < s.Length)
        {      
            dic[s[r]] = dic.GetValueOrDefault(s[r], 0) + 1;
            r++;

            while (dic.Values.Sum() - dic.Values.Max() > k)
            {
                dic[s[l]]--;
                l++;
            }
            result = Math.Max(result, r - l);
        }
        return result;
    }
}
